homeassistant/components/aladdin_connect/diagnostics.py

```diff
@@ -0,0 +1,31 @@
+"""Diagnostics support for Aladdin Connect."""
+from __future__ import annotations
+
+from typing import Any
+
+from AIOAladdinConnect import AladdinConnectClient
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+TO_REDACT = {"serial", "device_id"}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+
+    acc: AladdinConnectClient = hass.data[DOMAIN][config_entry.entry_id]
+
+    diagnostics_data = {
+        "data": {
+            "doors": async_redact_data(acc.doors, TO_REDACT),
+        },
+    }
```

Comment:
```suggestion
    diagnostics_data = {
        "doors": async_redact_data(acc.doors, TO_REDACT),
    }
```

Reply:
You could consider to use `from tests.components.diagnostics import get_diagnostics_for_config_entry` to get the full diagnostics output as the user would receive it.

Reply:
> You could consider to use `from tests.components.diagnostics import get_diagnostics_for_config_entry` to get the full diagnostics output as the user would receive it.

I'm a bit confused by this comment.  Isn't this what it's doing?
https://github.com/home-assistant/core/blob/d45b33e55a4c48dd938e91df00ee37650c255ea0/tests/components/aladdin_connect/test_diagnostics.py#L11

https://github.com/home-assistant/core/blob/d45b33e55a4c48dd938e91df00ee37650c255ea0/tests/components/aladdin_connect/test_diagnostics.py#L42

Reply:
Yes. Sorry! Looking at too much at the same time.
I think it's done so if you set it for review.